timeseries/src/autogluon/timeseries/models/abstract/abstract_timeseries_model.py

```diff
@@ -369,6 +527,31 @@ def _preprocess_time_limit(self, time_limit: float) -> float:
 
         return time_limit
 
+    def _fit(  # type: ignore
+        self,
+        train_data: TimeSeriesDataFrame,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: Optional[float] = None,
+        num_cpus: Optional[int] = None,
+        num_gpus: Optional[int] = None,
+        verbosity: int = 2,
+        **kwargs,
+    ) -> None:
+        """Private method for `fit`. See `fit` for documentation of arguments. Apart from
+        the model training logic, `fit` additionally implements other logic such as keeping
+        track of the time limit, etc.
+        """
+        # TODO: Make the models respect `num_cpus` and `num_gpus` parameters
+        raise NotImplementedError
+
+    def _check_fit_params(self):
```

Comment:
Let's add a TODO to perform this check inside `AbstractTimeSeriesModel.fit()` instead of calling it explicitly in every subclass (like [here](https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/models/gluonts/abstract_gluonts.py#L484))